solang-parser/src/solidity.lalrpop

```diff
@@ -249,6 +254,14 @@ VariableDefinition: Box<VariableDefinition> = {
     }
 }
 
+TypeDefinition: Box<TypeDefinition> = {
+    <doc:DocComments> <l:@L> "type" <name:SolIdentifier> "is" <ty:ElementaryValueType> <r:@R> ";" => {
```

Comment:
It would be nice if the parser could parse:
```
contract c {
	type foo is int[2];
}
```
Then we could produce a nice error message say "int[2]" is not an elementary value type (why it is not called a primitive type like every other language, I don't know!).

```
<doc:DocComments> <l:@L> "type" <name:SolIdentifier> "is" <ty:Precedence0> <r:@R> ";" => {
```
Will do the trick (I think).

Reply:
yeah totally agree, fixed